### PR TITLE
[ENH] `MeanSquaredLogarithmicError` (MSLE) metric (#9139)

### LINKED ISSUE #9139: [ENH] Add MeanSquaredLogarithmicError (MSLE) metric
**Is your feature request related to a problem? Please describe.**
`sktime` currently lacks **Mean Squared Logarithmic Error (MSLE)** and its root variant **RMSLE**.

These metrics are industry standards (common in Kaggle competitions and sales forecasting) for data with exponential growth patterns, where:
1.  Errors on large values should not dominate the metric.
2.  Under-prediction is penalized more heavily than over-prediction.
3.  Relative error matters more than absolute error.

**Describe the solution you'd like**
I propose implementing `MeanSquaredLogError` in `sktime.performance_metrics.forecasting`.

**Implementation Plan:**
- Add `mean_squared_log_error` functional logic using `np.log1p`.
- Add `MeanSquaredLogError` class wrapper inheriting from `BaseForecastingErrorMetric`.
- Support `square_root=True` to calculate **RMSLE**.
- Implement `_evaluate_by_index` to return the vector of squared log errors for time-series plotting/analysis.

**Describe alternatives you've considered**
Users currently have to manually log-transform their data, use MSE, and then try to interpret the results, or wrap `sklearn.metrics.mean_squared_log_error`. A native `sktime` implementation ensures proper handling of `multioutput`, `horizon_weight`, and time indices.

**Will you work on this?**
Yes, I have the implementation ready.

### PR DESCRIPTION
**Pull request by alphaleporus**
#### Reference Issues/PRs
Fixes #9139

#### What does this implement/fix? Explain your changes.
This PR implements **Mean Squared Logarithmic Error (MSLE)** and **Root Mean Squared Logarithmic Error (RMSLE)**.

These metrics are essential for forecasting tasks where the target variable spans several orders of magnitude or follows an exponential growth curve (e.g., sales, population growth), and where relative error is more important than absolute error.

**Implementation Details:**
- Added `mean_squared_log_error` to `sktime/performance_metrics/forecasting/_functions.py` using `np.log1p` for numerical stability.
- Added `MeanSquaredLogError` class in `sktime/performance_metrics/forecasting/_msle.py`.
- Implemented `_evaluate_by_index` to return the vector of squared log errors, adhering to the sktime metric architecture.
- Exposed via `sktime/performance_metrics/forecasting/__init__.py`.

**Math:**
$$\text{MSLE} = \frac{1}{n} \sum_{i=1}^{n} (\log(1 + y_i) - \log(1 + \hat{y}_i))^2$$

If `square_root=True` (RMSLE):
$$\text{RMSLE} = \sqrt{\text{MSLE}}$$

#### Does your contribution introduce a new dependency?
No.

#### Did you add any tests for the change?
Added `get_test_params` (testing both standard and `square_root=True` cases) to ensure coverage by the standard test suite. Verified logic locally against manual calculations.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) (will do post-merge)
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].

##### For new estimators
- [ ] I've added the estimator to the API reference
- [x] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.

### CODE REVIEW

**Comment by alphaleporus** [comment]
I noticed run-notebook-examples failed with an **UnpicklingError** related to _**pytorch-forecasting and torch.load**_ security settings (PyTorch 2.6 change). This seems unrelated to the MSLE metric implementation.

All relevant functional tests (including test-nosoftdeps and full matrix) passed. ✅

Ready for review!

**Comment by alphaleporus** [comment]
I noticed unrelated CI failures in test_mlflow (SQLite concurrency issue) and run-notebook-examples (PyTorch 2.6 unpickling security change).

The core logic tests for MSLE have passed successfully. ✅ Ready for review!

File: sktime/performance_metrics/forecasting/_msle.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        y_true_np = np.maximum(y_true_np, 0)
+        y_pred_np = np.maximum(y_pred_np, 0)
+
+        y_true_log = np.log1p(y_true_np)
+        y_pred_log = np.log1p(y_pred_np)
+
+        squared_log_error = np.square(y_true_log - y_pred_log)
+
+        # Reconstruct Pandas container if input was Pandas to support aggregation
+        if isinstance(y_true, (pd.DataFrame, pd.Series)):
```
Comment (fkiraly) [inline sktime/performance_metrics/forecasting/_msle.py]: I think this is unnecessary, since there is only one thing that `y_true` can be afaik at this moment - a `pd.DataFrame`

File: sktime/performance_metrics/forecasting/_msle.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        squared_log_error = self._get_weighted_df(squared_log_error, **kwargs)
+
+        return self._handle_multioutput(squared_log_error, self.multioutput)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator."""
+        params1 = {}
+        params2 = {"square_root": True}
+        params3 = {"multioutput": "raw_values"}
```
Comment (fkiraly) [inline sktime/performance_metrics/forecasting/_msle.py]: `multioutput` does not need to be set explicitly, it is changed by the tests.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great! Only minor change requests.

File: sktime/performance_metrics/forecasting/_msle.py (hunk 1 above)
Comment (alphaleporus) [reply sktime/performance_metrics/forecasting/_msle.py]: Done. I've removed the `isinstance` check and am now unconditionally reconstructing the DataFrame using `y_true.index` and `y_true.columns`.

File: sktime/performance_metrics/forecasting/_msle.py (hunk 2 above)
Comment (alphaleporus) [reply sktime/performance_metrics/forecasting/_msle.py]: Removed `params3` and the redundant `multioutput` setting as suggested.

**Comment by alphaleporus** [comment]
Changes applied! Ready for merge.

**Review by fkiraly — APPROVED** [review]
Looks good to me now! Thanks!

**Comment by alphaleporus** [comment]
Hi @fkiraly, just a quick follow-up on this one! The PR has been approved, and all CI checks are green. Is there anything else needed from my side to get this merged?

**Comment by fkiraly** [comment]
oh, sorry - we forgot to merge.

**Comment by alphaleporus** [comment]
@all-contributors please add @alphaleporus for code
